### DISCUSSION TITLE
Remove MV2 support in @types/chrome

### DISCUSSION THREAD

**Discussion by nicolas377** [DefinitelyTyped/DefinitelyTyped#55600] [_Misc]
Since MV3 is out now, developers should be creating their extensions in MV3, and a nice way to push devs to MV3 is to remove MV2 types. I didn't make this a PR so that we can discuss whether or not to do that.

**Comment by erwanjugand** [comment]
Hello @nicolas377 

I’m in favor of removing support for Manifest V2.

According to [this documentation](https://developer.chrome.com/docs/extensions/develop/migrate/mv2-deprecation-timeline#october_9th_2024_an_update_on_manifest_v2_phase-out), it has not been possible to publish Manifest V2 extensions on the Chrome Web Store since June 2022.

I think there are no more MV2 extensions currently in development.

(ping @fregante)

**Reply by nicolas377** [reply]
When I originally wrote this MV2 was starting to head towards it's EOL, but now that MV2 is long gone, I definitely see the value in removing the support. My only question is this: what would removing MV2 mean in terms of direct changes?

**Reply by erwanjugand** [reply]
There are 3 namespaces for Manifest V2: [chrome.browserAction](https://developer.chrome.com/docs/extensions/mv2/reference/browserAction), [chrome.declarativeWebRequest](https://developer.chrome.com/docs/extensions/mv2/reference/declarativeWebRequest), and [chrome.pageAction](https://developer.chrome.com/docs/extensions/mv2/reference/pageAction).

The `ManifestV2` type can also be removed (And merge `ManifestBase` and `ManifestV3)`.

We can also remove terms like `(MV3 only)`, `in Manifest V3`, `in Manifest V3 or later`, `@since MV3`, `@requires MV3+`, `MV3`, etc.

**Reply by jakebailey** [reply]
Isn't this option still available for enterprises, and available in Edge, and available still in other forks?

**Reply by fregante** [reply]
I'm all for dropping obsolete code, but MV2 extensions technically still run in Chrome 133, so it might be premature to drop the types. 

We could consider the idea that "MV2 users should stick to `@types/chrome@0.0.302`", but such advice would be lost on this repo.

**Reply by erwanjugand** [reply]
We could also use SEMVER and move to `0.1.x` to highlight a breaking change and create a CHANGELOG.md file. (If it's possible)

**Reply by erwanjugand** [reply]
> Starting in June, the branch for Chrome 139 will begin, in which support for Manifest V2 extensions will be removed from Chrome. Unlike the previous changes to disable Manifest V2 extensions which gradually rolled out to users, this change will impact all users on Chrome 139 at once. As a result, **Chrome 138 is the final version of Chrome to support Manifest V2 extensions** (when paired with the ExtensionManifestV2Availability key) 

https://developer.chrome.com/docs/extensions/develop/migrate/mv2-deprecation-timeline#march_31st_2025_an_update_on_manifest_v2_phase-out

**Reply by erwanjugand** [reply]
I created a draft PR (https://github.com/DefinitelyTyped/DefinitelyTyped/pull/73300) to review the changes it would involve (The tests are failing because of `@types/naver-whale` 😕)

Here are the occurrences of `MV2` found in the documentation:
<img width="552" height="758" alt="image" src="https://github.com/user-attachments/assets/a4f354c2-1d63-4189-a8aa-4276cac5c07e" />

**Reply by erwanjugand** [reply]
The documentation just updated all pages to remove the mention:

"Promises are supported in Manifest V3 and later, but callbacks are provided for backward compatibility. You cannot use both in the same function call. The promise resolves with the same type that is passed to the callback."

Watch diff [here](https://github.com/erwanjugand/chrome-extension-docs-checker/commit/a8de96362cf5d080e1e1a0cfe1f21b3d70be3916#diff-aa07d183bd745967368638dc7e6147d36b7c3fdeaf05fbc5ba23d0146ead9242)

**Comment by erwanjugand** [comment]
**Prerequisites for dropping Manifest V2:**
- Update all types
- Check store support

**Namespaces to check/update:** 
- [x] https://github.com/DefinitelyTyped/DefinitelyTyped/pull/71288
- [x] https://github.com/DefinitelyTyped/DefinitelyTyped/pull/72849
- [x] https://github.com/DefinitelyTyped/DefinitelyTyped/pull/73512
- [x] https://github.com/DefinitelyTyped/DefinitelyTyped/pull/70624
- [x] https://github.com/DefinitelyTyped/DefinitelyTyped/pull/72131
- [x] https://github.com/DefinitelyTyped/DefinitelyTyped/pull/73998
- [x] https://github.com/DefinitelyTyped/DefinitelyTyped/pull/73298
- [x] https://github.com/DefinitelyTyped/DefinitelyTyped/pull/72718
- [x] https://github.com/DefinitelyTyped/DefinitelyTyped/pull/73299
- [x] https://github.com/DefinitelyTyped/DefinitelyTyped/pull/72092
- [x] https://github.com/DefinitelyTyped/DefinitelyTyped/pull/72253
- [x] https://github.com/DefinitelyTyped/DefinitelyTyped/pull/73395
- [x] https://github.com/DefinitelyTyped/DefinitelyTyped/pull/72139
- [x] https://github.com/DefinitelyTyped/DefinitelyTyped/pull/73526
- [x] https://github.com/DefinitelyTyped/DefinitelyTyped/pull/74000
- [x] https://github.com/DefinitelyTyped/DefinitelyTyped/pull/73699
- [x] https://github.com/DefinitelyTyped/DefinitelyTyped/pull/73413
- [x] https://github.com/DefinitelyTyped/DefinitelyTyped/pull/73473
- [x] https://github.com/DefinitelyTyped/DefinitelyTyped/pull/73475
- [x] https://github.com/DefinitelyTyped/DefinitelyTyped/pull/73504
- [x] https://github.com/DefinitelyTyped/DefinitelyTyped/pull/73476
- [x] https://github.com/DefinitelyTyped/DefinitelyTyped/pull/72261
- [x] https://github.com/DefinitelyTyped/DefinitelyTyped/pull/71906
- [x] ✅ dom
- [x] https://github.com/DefinitelyTyped/DefinitelyTyped/pull/72917
- [x] https://github.com/DefinitelyTyped/DefinitelyTyped/pull/73471
- [x] https://github.com/DefinitelyTyped/DefinitelyTyped/pull/70591
- [x] https://github.com/DefinitelyTyped/DefinitelyTyped/pull/73439
- [x] ✅ enterprise.networkingAttributes
- [x] https://github.com/DefinitelyTyped/DefinitelyTyped/pull/72977
- [x] https://github.com/DefinitelyTyped/DefinitelyTyped/pull/72973
- [x] https://github.com/DefinitelyTyped/DefinitelyTyped/pull/73523
- [x] https://github.com/DefinitelyTyped/DefinitelyTyped/pull/72850
- [x] https://github.com/DefinitelyTyped/DefinitelyTyped/pull/73528
- [x] https://github.com/DefinitelyTyped/DefinitelyTyped/pull/72156
- [x] https://github.com/DefinitelyTyped/DefinitelyTyped/pull/73531
- [x] https://github.com/DefinitelyTyped/DefinitelyTyped/pull/72990
- [x] https://github.com/DefinitelyTyped/DefinitelyTyped/pull/73535
- [x] https://github.com/DefinitelyTyped/DefinitelyTyped/pull/73536
- [x] https://github.com/DefinitelyTyped/DefinitelyTyped/pull/72070
- [x] https://github.com/DefinitelyTyped/DefinitelyTyped/pull/73565
- [x] https://github.com/DefinitelyTyped/DefinitelyTyped/pull/73661
- [x] https://github.com/DefinitelyTyped/DefinitelyTyped/pull/73591
- [x] https://github.com/DefinitelyTyped/DefinitelyTyped/pull/73667
- [x] https://github.com/DefinitelyTyped/DefinitelyTyped/pull/73776
- [x] https://github.com/DefinitelyTyped/DefinitelyTyped/pull/72598
- [x] https://github.com/DefinitelyTyped/DefinitelyTyped/pull/73701
- [x] https://github.com/DefinitelyTyped/DefinitelyTyped/pull/73739
- [x] https://github.com/DefinitelyTyped/DefinitelyTyped/pull/73999
- [x] https://github.com/DefinitelyTyped/DefinitelyTyped/pull/73785
- [x] https://github.com/DefinitelyTyped/DefinitelyTyped/pull/71278
- [x] https://github.com/DefinitelyTyped/DefinitelyTyped/pull/72977
- [x] https://github.com/DefinitelyTyped/DefinitelyTyped/pull/70569
- [x] https://github.com/DefinitelyTyped/DefinitelyTyped/pull/73719
- [x] https://github.com/DefinitelyTyped/DefinitelyTyped/pull/70664
- [x] https://github.com/DefinitelyTyped/DefinitelyTyped/pull/70797
- [x] https://github.com/DefinitelyTyped/DefinitelyTyped/pull/71289
- [x] https://github.com/DefinitelyTyped/DefinitelyTyped/pull/73761
- [x] https://github.com/DefinitelyTyped/DefinitelyTyped/pull/73783
- [x] https://github.com/DefinitelyTyped/DefinitelyTyped/pull/73393
- [x] https://github.com/DefinitelyTyped/DefinitelyTyped/pull/73023
- [x] https://github.com/DefinitelyTyped/DefinitelyTyped/pull/73763
- [x] https://github.com/DefinitelyTyped/DefinitelyTyped/pull/73784
- [x] https://github.com/DefinitelyTyped/DefinitelyTyped/pull/73794
- [x] https://github.com/DefinitelyTyped/DefinitelyTyped/pull/74072
- [x] https://github.com/DefinitelyTyped/DefinitelyTyped/pull/73797
- [x] https://github.com/DefinitelyTyped/DefinitelyTyped/pull/71042
- [x] https://github.com/DefinitelyTyped/DefinitelyTyped/pull/73798
- [x] https://github.com/DefinitelyTyped/DefinitelyTyped/pull/73818
- [x] https://github.com/DefinitelyTyped/DefinitelyTyped/pull/70566
- [x] https://github.com/DefinitelyTyped/DefinitelyTyped/pull/73802
- [x] https://github.com/DefinitelyTyped/DefinitelyTyped/pull/72665
- [x] https://github.com/DefinitelyTyped/DefinitelyTyped/pull/72974
- [x] https://github.com/DefinitelyTyped/DefinitelyTyped/pull/73826
- [x] https://github.com/DefinitelyTyped/DefinitelyTyped/pull/72243
- [x] https://github.com/DefinitelyTyped/DefinitelyTyped/pull/71692
- [x] https://github.com/DefinitelyTyped/DefinitelyTyped/pull/73876
- [x] https://github.com/DefinitelyTyped/DefinitelyTyped/pull/73839
- [x] https://github.com/DefinitelyTyped/DefinitelyTyped/pull/73873
- [x] https://github.com/DefinitelyTyped/DefinitelyTyped/pull/73835
- [x] https://github.com/DefinitelyTyped/DefinitelyTyped/pull/72752
- [x] https://github.com/DefinitelyTyped/DefinitelyTyped/pull/73875
- [x] https://github.com/DefinitelyTyped/DefinitelyTyped/pull/72905
- [x] https://github.com/DefinitelyTyped/DefinitelyTyped/pull/72921
- [x] https://github.com/DefinitelyTyped/DefinitelyTyped/pull/74847

**Stores EoL MV2:**
- Chrome Web Store: [Juin 2022](https://developer.chrome.com/docs/extensions/develop/migrate/mv2-deprecation-timeline#june_2022_chrome_web_store_-_no_new_private_extensions)
- Microsoft Edge Add-ons: [TBD](https://learn.microsoft.com/en-us/microsoft-edge/extensions/developer-guide/manifest-v3)
- Opera Add-ons: [TBD](https://blogs.opera.com/news/2025/09/mv2-extensions-opera/)

**Reply by erwanjugand** [reply]
FYI, Edge has updated its timeline and will start showing deprecation warnings around Early 2027 for enterprise.
